client/extensions/woocommerce/app/products/product-form-variations-card.js

```diff
@@ -22,7 +22,21 @@ const ProductFormVariationsCard = ( {
 } ) => {
 	const handleToggle = () => {
 		if ( 'variable' !== product.type ) {
-			editProduct( product, { type: 'variable', dimensions: {} } );
+			// Reset any 'simple' product settings that have been filled out,
```

Comment:
We had decided at some point to clear out variation types when the type gets switched back to simple, so if we do that, we should do the same in reverse.

However - I agree with you that it's a little too easy to lose state by toggling that. I'm not sure what the answer is though, because we don't want to pass that extra data to the API.